overlord/snapstate/component.go

```diff
@@ -373,7 +373,22 @@ func doInstallComponent(st *state.State, snapst *SnapState, compSetup ComponentS
 
 	componentTS.beforeLinkTasks = append(componentTS.beforeLinkTasks, prepare)
 
-	if fromStore {
+	// if we're installing a component from the store, then we need to validate
```

Comment:
CompPath == "" here means from the store right?

Reply:
I'd say that it means that we expect it to have a valid `snap-resource-pair` assertion somewhere. A bit above there is also `fromStore := compSetup.CompPath == "" && !revisionIsPresent`, that to me means "we need to download from the store".

Reply:
Currently the only time that no path is provided is when we’re either downloading from the store, or we’re doing a revert (during a revert we don’t provide a download info or a path). That’s why I checked for the revert flag too.

Reply:
for later: it feels like maybe we should rename for clarity fromStore to needsDowload ?and define a different fromStore ? we would maybe do the same for doInstall?